### PR TITLE
Solution to #115, Updating VoterTest.java to Junit5

### PR DESCRIPTION
**Pull request by Kelzihar**
- Solves #115 
- Changed the dependencies to be concurrent with Junit5
- Addition of comments for each test

### CODE REVIEW

**Review by jody — CHANGES_REQUESTED** [review]
Given that "assertEquals" was available at the time VoterTest was created, consider the original choice to use "assertThat" rather than "assertEquals".

Might it be better to address this by using [Hamcrest](http://hamcrest.org)? That would maintain the original test code and changing the imports. That also provides for future use of the more advanced matchers.

**Comment by Kelzihar** [comment]
I believe I changed up the test enough to include hamcrest instead of the assertEquals

**Comment by jody** [comment]
Since this doesn't fully address any existing issue, there's nothing to associate with this pull request.

To make a partial contribution to an issue, first create a sub-issue for the original issue and indicate inter-dependencies (that the original issue depends on the new issue; that the new issue is a dependent of the original issue).

File: src/swdmt/redistricting/VoterTest.java [hunk 1]
```diff
@@ -23,19 +31,29 @@ public void affiliationAttributeNullTest() {
...
+        });
     }
 
+    /**
+    * When passed various location attributes they match expected values
+    */
     @Test
     public void locationAttributeTest() {
         Voter voter1 = new Voter(Party.NONE, new Location(0, 0));
-        assertThat(voter1.location(), is(new Location(0, 0)));
```
Comment (jody) [inline src/swdmt/redistricting/VoterTest.java:34]: Why was the actual ordering of construction and testing changed?

Note that this change of ordering can result in the test (original line 34) not being executed if the invocation of the constructor (original lines 35-37) fails.

This appears to be a potential behavioral change to the test rather than just a refactoring to address upgrading from JUnit 4 to JUnit 5.

File: src/swdmt/redistricting/VoterTest.java [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
+    * Voter's party affiliation should match Party. None when passed Party.None
+    */
     @Test
     public void affiliationAttributeSpecifiedTest() {
         Voter voter1 = new Voter(Party.NONE, new Location(0, 0));
         assertThat(voter1.affiliation(), is(Party.NONE));
     }
 
+    /**
+    * Voter's party affiliation should be Party. None when passed null
```
Comment (jody) [inline src/swdmt/redistricting/VoterTest.java:24]: Confusing description. Perhaps a typo?

File: src/swdmt/redistricting/VoterTest.java [hunk 3]
```diff
@@ -1,20 +1,28 @@
...
 /**
  * Tests for Voter objects.
  *
  * @author  Dr. Jody Paul
- * @version 20191006
+ * @version 20211120
  */
 public class VoterTest {
+    /**
+    * Voter's party affiliation should match Party. None when passed Party.None
```
Comment (jody) [inline src/swdmt/redistricting/VoterTest.java:15]: Confusing description. Perhaps typo?

File: src/swdmt/redistricting/VoterTest.java [hunk 4]
```diff
@@ -23,19 +31,29 @@ public void affiliationAttributeNullTest() {
         assertThat(voter2.affiliation(), is(Party.NONE));
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void invalidLocationTest() {
-        Voter voter1 = new Voter(Party.NONE, null);
+    /**
+    * When passed a null location should throw an illegal argument exception
```
Comment (jody) [inline src/swdmt/redistricting/VoterTest.java:35]: Javadoc comments must begin with a sentence.

File: src/swdmt/redistricting/VoterTest.java [hunk 5]
```diff
@@ -23,19 +31,29 @@ public void affiliationAttributeNullTest() {
...
+    */
+    @Test
+    public void invalidLocationTest() throws IllegalArgumentException {
+        assertThrows(IllegalArgumentException.class, () -> {
+            Voter voter1 = new Voter(Party.NONE, null);
+        });
     }
 
+    /**
+    * When passed various location attributes they match expected values
```
Comment (jody) [inline src/swdmt/redistricting/VoterTest.java:45]: Must begin with a sentence.

File: src/swdmt/redistricting/VoterTest.java [hunk 6]
```diff
@@ -23,19 +31,29 @@ public void affiliationAttributeNullTest() {
...
-        assertThat(voter1.location(), is(new Location(0, 0)));
         Voter voter2 = new Voter(null,
                                  new Location(Integer.MIN_VALUE,
                                               Integer.MAX_VALUE));
-        assertThat(voter2.location().xCoordinate(), is(Integer.MIN_VALUE));
-        assertThat(voter2.location().yCoordinate(), is(Integer.MAX_VALUE));
+        assertAll("locations",
+            () -> assertThat(voter1.location(), is(new Location(0, 0))),
+            () -> assertThat(voter2.location().xCoordinate(), is(Integer.MIN_VALUE)),
+            () -> assertThat(voter2.location().yCoordinate(), is(Integer.MAX_VALUE))
```
Comment (jody) [inline src/swdmt/redistricting/VoterTest.java:56]: Consider reverting to original. Unclear why this change is being made.

**Review by jody — CHANGES_REQUESTED** [review]

**Comment by Kelzihar** [comment]
I believe I have gone through and made the requested changes, let me know if there still appear to be issues with the comments

**Review by jody — CHANGES_REQUESTED** [review]
Changes all look good!

Please squash commits (currently 15) to a single commit with appropriate comment.

**Comment by Kelzihar** [comment]
I believe all commits have been squashed

**Review by jody — APPROVED** [review]
Great! Thanks!!
